source/LocalizationManager.cs

```diff
@@ -113,16 +119,17 @@ private static Localizer CreateLocalizer (Localizer fallback, string culture, IT
 				}
 				catch (Exception e)
 				{
-					throw new TemplateSetLoadException(culture, setname, e);
+					throw new TemplateSetLoadException(culture, provider.Name, e);
```

Comment:
entire `try {} catch () {}` needs to go into `CompatibilityTemplateProvider.Load()`, since that's where culture-specific template set loading responsibility is moved.